plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -195,6 +212,13 @@ public ConnectorPageSource createPageSource(
                 split.getFileModifiedTime());
     }
 
+    private static boolean dataColumnsProjected(List<DeltaLakeColumnHandle> columns)
```

Comment:
I am not sure I understand this naming, by its name I would assume that you map these columns to some other types or something like that, but you simply check if there is any column of type REGULAR